app/controllers/provider_interface/candidates_controller.rb

```diff
@@ -21,11 +21,13 @@ def impersonate
   private
 
     def verify_provider_association(candidate:, providers:)
-      provider_ids_from_candidate = candidate.application_forms
-                                              .map(&:application_choices).flatten
-                                              .map(&:provider).map(&:id).uniq
+      application_choices = candidate.application_forms.map(&:application_choices).flatten
+      candidate_providers = application_choices.map(&:provider) + application_choices.map(&:accredited_provider)
+      provider_ids_from_candidate = candidate_providers.compact.map(&:id).uniq
+      auth = ProviderAuthorisation.new(actor: current_provider_user)
 
-      providers.any? { |provider| provider_ids_from_candidate.include? provider.id }
+      providers.any? { |provider| provider_ids_from_candidate.include? provider.id } &&
```

Comment:
Just thought: if we're going with this permissions requirement, does the auth do the provider membership check for us?

Reply:
Good point, it does call `GetApplicationChoicesForProvider` which seems to be doing the same thing.